src/NuGet.Clients/NuGet.CommandLine/Program.cs

```diff
@@ -68,6 +69,8 @@ public static int MainCore(string workingDirectory, string[] args)
             }
 
             var console = new Common.Console();
+            Logger.Instance = console;
```

Comment:
I counted 11 instances of Program.Main in tests NuGetConfigCommandTest, NuGetInstallCommandTest, NuGetListCommandTest. Apparently this technique allows significant time saving as opposed to a conventional exe run. And when declared as async these tests DO run in parallel. So having static member conceals potential danger here.

One alternative solution could be using CallContext (see HttpContext.Current).

BUT as this doesn't imply any immediate risk on product user scenarios, I'd suggest simply adding a TODO comment to re-implement this later.

Does it make sense?